### PR TITLE
Allow secret names for user plugins to contain template language

### PR DESCRIPTION
**Pull request by robbie-demuth**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:

Our organization deploys a Helm chart that actually has a dependency on the `artifactory-ha` chart. Now that the Artifactory chart supports user plugins, we'll create the secrets containing the user plugins in our 'wrapper' chart. Since the names of the secrets are named according to the output of template functions, we don't have a hardcoded secret name that we can pass to Artifactory. This PR updates the Artifactory chart to allow the names of externally provided secrets to contain template language

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

FYI @jainishshah17 @eldada - This is the change I was suggesting in https://github.com/jfrog/charts/pull/278#discussion_r270626240

### CODE REVIEW

File: stable/artifactory-ha/README.md [hunk 1]
```diff
@@ -440,6 +440,7 @@ artifactory:
   userPluginSecrets:
     - archive-old-artifacts
     - webhook
+    - '{{ template "my-chart.fullname" . }}'
```
Comment (eldada) [inline stable/artifactory-ha/README.md]: Can you add a short explanation of this use case. This mix in the example is confusing.

File: stable/artifactory/README.md [hunk 2]
```diff
@@ -353,6 +353,7 @@ artifactory:
   userPluginSecrets:
     - archive-old-artifacts
     - webhook
+    - '{{ template "my-chart.fullname" . }}'
```
Comment (eldada) [inline stable/artifactory/README.md]: Can you add a short explanation of this use case. This mix in the example is confusing.

File: stable/artifactory-ha/README.md [hunk 3]
```diff
@@ -447,6 +447,14 @@ You can now pass the created `plugins.yaml` file to helm install command to depl
 helm install --name artifactory-ha -f plugins.yaml jfrog/artifactory-ha
 ```
 
+Alternatively, you may be in a situation in which you would like to create a secret in a Helm chart that depends on this chart. In this scenario, the name of the secret is likely dynamically generated via template functions, so passing a statically named secret isn't possible. In this case, the chart supports evaluating strings as templates via the [`tpl`](https://helm.sh/docs/charts_tips_and_tricks/#using-the-tpl-function) function - simply pass the raw string containing the templating language used to name your secret as a value instead by adding the following to your chart's `values.yaml` file:
```
Comment (robbie-demuth) [inline stable/artifactory-ha/README.md]: Is this more clear, @eldada? I've added the same blob in the `artifactory` README

Comment (eldada) [reply stable/artifactory-ha/README.md]: Perfect. Thx!

**Comment by eldada** [comment]
@robbie-demuth - please rebase and update versions. Once conflicts are resolved, I'll approve and merge.

**Comment by rimusz** [comment]
nice one 👍

**Comment by robbie-demuth** [comment]
Done, @eldada!

**Pull request merged**